### DISCUSSION TITLE
Dynamic SMS cost landscape

### DISCUSSION THREAD

**Discussion by JoshL22** [RangeShifter/RangeshiftR-tutorials#5] [Q&A] [answered]
Hi all, 

I would like to use dynamic SMS cost values for my landscape (ideally 3 different sets). I have read that this is possible in batch mode, does this also include RangeshifteR ? If possible, how do I adapt the code to include different cost values ? 

My transfer line is as follows,

Transfer = SMS(PR=9,PRMethod = 2, DP=3.35, Costs = c(1000,1,1,1,1,1,500,1000,1000,900,800), StepMort = c(0.008,0.0001,0.0001,0.0001,0.0001,0.0001,0.0003,0.0007,0.08,0.06,0.04)

Many thanks, 

Josh

**Comment by damariszurell** [comment, marked as answer]
Hi there, 

yes you can do this in RangeShiftR. Your code would look similar to the one you have now. Simply your landscape is changing with your patch being assigned different land types in the different dynamic landscapes. When you look at tutorial 3 (https://rangeshifter.github.io/RangeshiftR-tutorials/tutorial_3.html), there are also the land types changing. In the transfer module you would set the costs that apply to all possible land types that you have across all your dynamic landscapes.

Best,
Damaris

**Reply by JoshL22** [reply]
Ok thank you for the explanation, I think I have an idea of what to do now.
